app/models/like.rb

```diff
@@ -0,0 +1,10 @@
+class Like < ApplicationRecord
+  belongs_to :user
+  belongs_to :post
+
+  def self.update_likes_counter(post_id)
+    post = Post.find(post_id)
+    likes_count = post.likes.count
+    post.update(likes_counter: likes_count)
+  end
```

Comment:
- [ Required ] I think it would be better if you used one of the [Active Rails callback](https://guides.rubyonrails.org/active_record_callbacks.html) e.g the `after_save` to call the `#update_likes_counter` method every time a new like has been saved to the database, then make the `update_like_counter` method `private`. This ensures that the `likes_counter` field is incremented anytime a like is added to a post.

Reply:
Fixed all of them @Felix45 💯 . Thank you.